pkg/arvo/app/launch.hoon

```diff
@@ -40,36 +41,55 @@
       path-to-tile=(map path @tas)
       first-time=?
   ==
++$  state-two
+  $:  %2
+      tiles=(set tile:launch)
+      data=tile-data:launch
+      path-to-tile=(map path @tas)
+      first-time=?
+      setup-who=_%.y
+  ==
 ::
 +$  card  card:agent:gall
 --
 ::
-=|  state-one
+=|  state-two
 =*  state  -
 %-  agent:dbug
 ^-  agent:gall
 |_  bol=bowl:gall
 +*  this  .
     def   ~(. (default-agent this %|) bol)
+    launch-who  [%pass /who %arvo %e %serve [~ /who] q.byk.bol /gen/who/hoon ~]
 ++  on-init
   ^-  (quip card _this)
-  :_  this(state *state-one)
-  [%pass / %arvo %e %connect [~ /] %launch]~
+  :_  this(state *state-two)
+  :~
+    launch-who
+    [%pass / %arvo %e %connect [~ /] %launch]
+  ==
 ::
 ++  on-save  !>(state)
 ::
 ++  on-load
   |=  old=vase
   ^-  (quip card _this)
   =/  old-state  !<(versioned-state old)
-  :-  ~
   ?-    -.old-state
       %0
+    :-  ~[launch-who]
     %=  this
-      state  [%1 tiles.old-state data.old-state path-to-tile.old-state %.n]
+      state  [%2 tiles.old-state data.old-state path-to-tile.old-state %.n %.y]
     ==
   ::
-      %1  this(state old-state)
+      %1
+    :-  ~[launch-who]
+    %=  this
+      state  [%2 tiles.old-state data.old-state path-to-tile.old-state first-time.old-state %.y]
```

Comment:
Since the only things used here are constants and things from `old-state`, you could

```suggestion
      state  [%2 tiles data path-to-tile first-time %.y]:old-state
```

Also applies to the `%0` case I guess.